majority-multisign/src/MajorityMultiSign/OnChain.hs

```diff
@@ -69,19 +69,28 @@ hasNewSignatures (UpdateKeysAct keys) MajorityMultiSignDatum {signers} ctx =
 {-# INLINEABLE hasCorrectToken #-}
 hasCorrectToken :: MajorityMultiSignValidatorParams -> ScriptContext -> MajorityMultiSignDatum -> Bool
 hasCorrectToken MajorityMultiSignValidatorParams {asset} ctx expectedDatum =
-  traceIfFalse "Couldn't find asset" (isJust assetTxOut)
```

Comment:
Could this all have been achieved with simply
`traceIfFalse "Couldn't find asset" (isJust $ assetTxOut >>= txOutDatumhash)`
?

Unsure if the added script size (given this script will already be quite large, as its plutus) is worth the added verbosity of errors.

Reply:
Yes, it definitely would. Sorry, I forgot about script size, my usual Haskell reflexes kicked in.
Do you think that size growth is big because of more string literals or more pattern-matches?

Reply:
This (https://github.com/mlabs-haskell/majority-multisign/pull/40/commits/32b6d9ef7bc8dacd753e2907ba234e25d6cdf4dc) might reduce number of pattern-matches, if compiler will see that `traceIfNothing` does not change the tag

Reply:
My point with the above comment was just that I had assumed changing that `isJust` check would solve the entire problem. The rest of this PR would then simply be a refactor for better error logging. Its a question of if that is worth it. My team doesn't need this project anymore, so it really depends on how close you are to limits with your other scripts.

Reply:
Oh, I see. Well, if we have problems with performance, we'll come back to this. Thank you for review